### DISCUSSION TITLE
Advent of code with Gleam

### DISCUSSION THREAD

**Discussion by TristanCacqueray** [gleam-lang/gleam#1357] [General]
Hello folks,

Are you doing the adventofcode.com puzzle with Gleam?
I would like to learn the language with the new build tool and it would be useful to see your solutions.

Cheers,
-Tristan

**Comment by TristanCacqueray** [comment]
Well I somehow managed to finish the first puzzle, and for it is worth, here is my code: https://github.com/TristanCacqueray/advent-of-code/tree/master/2021

**Reply by TristanCacqueray** [reply]
I made it up to day 5 so far, going through a good chunk of the language tour and stdlib.
Thank you folks for making such a nice language, it is indeed very friendly and well thought out :-)

**Reply by lpil** [reply]
I'm glad you think so, thank you 💜

**Reply by rockwotj** [reply]
I have been going over advent of code (late) as an excuse to play with gleam, and I'd like to echo Tristan's comment - gleam is delightful! I'm really excited about this language. I like erlang, but have only toyed with it. I also happen to really like types, and I've been impressed with how clean and simple this language is, it's also much simpler to learn IMO than erlang. I think the language is really in a sweet spot and I'm excited to play with some of the OTP stuff with gleam.

Cheers and Happy New Years!

**Reply by lpil** [reply]
Thank you, that means a lot! Happy new year

**Comment by TanklesXL** [comment]
I was planning on it, if it helps/makes things easier for you, you can use the [template](https://github.com/TanklesXL/aoc_template) that I made. 
_note:_ currently requires gleam v0.18-rc2

**Comment by lpil** [comment]
I would quite like to, but I don't think I'm going to have the time unfortunately 😞 
Gotta get this next Gleam release out before anything else
